internal/resources/metal/project/bgp_config.go

```diff
@@ -56,7 +56,7 @@ func handleBGPConfigChanges(ctx context.Context, client *metalv1.APIClient, plan
 	bgpRemoved := plan.BGPConfig.IsNull() && !state.BGPConfig.IsNull()
```

Comment:
Arguably out of scope, since this is aiming to replicate prior behavior, but:  could we catch and reject BGP settings removal earlier, maybe via config validation or plan modification?

Reply:
Relates to:
* https://github.com/equinix/terraform-provider-equinix/issues/160
* https://github.com/packethost/terraform-provider-packet/issues/299

Reply:
I've introduced a custom planmodifier immutablelist to cover that. I have done it generic in case it could be reused in the future. That's will be the expected output:

![image](https://github.com/equinix/terraform-provider-equinix/assets/68897552/2e9e80a9-45ca-4b07-a01e-8263313643f5)